### ISSUE TITLE
Automatic sorting, after creating a new page/chapter

### ISSUE THREAD

**Issue by Biepa** [BookStackApp/BookStack#2065]
**Describe the feature you'd like**
To save the last used sorting on a book, so when a new page/chapter is created, the page/chapter is inserted at the right place and not at the end of the list.

**Describe the benefits this feature would bring to BookStack users**
No manual sorting after every new page/chapter.

**Additional context**
Another idea is a global option you can choose, for the default sort and then a "custom sorting" option on a book/chapter, so a manual sorting is still possible. But that's maybe a lot more work.

**Comment by Rumcajsev** [comment]
I agree. An example from my usage is having a new page for each retrospection, with date as a page name. It would be useful to automatically sort these - eg. by date of creation.

**Comment by Stoicmaverick** [comment]
I would really like to have the ability to order pages/books, as it would make finding specific pages much easier, paticularly on mobile, so you don't have to enlist your other hand to type out a search.

Maybe sorting options like Alphabetical, date accessed/created/edited, length. ect. I'm just a lowly soldering iron jockey, but I feel like this would be fairly simple to implement with low risk of bug introduction. I've seen several (MANY) requests for this, but I'd be willing to toss $10 in a bounty pool to have this implemented. (wait, can we start a bug/feature request bounty pool? I'm down.)

Also open in #180

**Comment by SteveDinn** [comment]
I'm planning to use a shelf / book as a blog, so it'd be super cool if I could automatically have books and pages sort in reverse-chronological order automatically.

**Comment by DanMundy** [comment]
I'd also like this.

The granular sorting options are _really good_ and I don't want this functionality to go away. I think it's a good default.

But there are some books where I want automatic sorting, and I want the chapters at the top. Currently I do the following steps _every single time I create a new page_, and it seems like something that I should be able to automate:

1. Click Sort
2. Click "Sort by name"
3. Click "Show chapters first"
4. Scroll right down to the bottom
5. Click Save

**Comment by ssddanbrown** [comment]
If anyone is really desperate for this, I've written up a logical theme system customization that can achieve this: https://gist.github.com/ssddanbrown/78553a39d36187476283a00fd67ed3dc

@DanMundy This follows your defined logic.

**Comment by ssddanbrown** [comment]
Thanks for the original request @Biepa, and thanks for the input from everyone else.

Within #5457 I have added "Sort Rules", which allows an admin to centrally define sort operations which can be applied as a default for new books, assigned to existing books by book editors, or be mass assigned via command line.

This will be part of the next feature release.
Since this should cover the fundamental request here, I'll go ahead and close this off.

**Comment by Biepa** [comment]
Hey @ssddanbrown,
I don't use Bookstack currently. Not because I didn't liked but the usecase wasn't there anymore.
But I just wanted to thank you for building Bookstack and keep improving it over all the years.
Will come back whenever the need comes up again. 👍
